src/renderer/dx/DxFontInfo.cpp

```diff
@@ -202,6 +202,63 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
     return face;
 }
 
+// Routine Description:
+// - Locates a suitable font face from the given information using more advanced match filtering on font sets only
+//   possible with Win10 and higher versions of DirectWrite
+// Arguments:
+// - fontCollection - The font collection to dig through
+// - localeName - The locale to search for appropriate fonts.
+// Return Value:
+// - Smart pointer holding interface reference for queryable font data.
+// - The pointer can be empty/null if there was a failure or we're below Win10 and need to fallback
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontInfo::_TryWin10FontMatchingLogic(gsl::not_null<IDWriteFontCollection*> fontCollection,
+                                                                                              std::wstring& localeName)
+{
+    Microsoft::WRL::ComPtr<IDWriteFontFace1> fontFace;
+
+    try
+    {
+        Microsoft::WRL::ComPtr<IDWriteFontCollection1> collection1;
+        THROW_IF_FAILED(fontCollection->QueryInterface(collection1.GetAddressOf()));
+
+        Microsoft::WRL::ComPtr<IDWriteFontSet> set;
+        THROW_IF_FAILED(collection1->GetFontSet(set.GetAddressOf()));
+
+        Microsoft::WRL::ComPtr<IDWriteFontSet1> set1;
+        THROW_IF_FAILED(set.As(&set1));
+
+        const std::array<DWRITE_FONT_PROPERTY, 3> fontProperties{
+            DWRITE_FONT_PROPERTY{ DWRITE_FONT_PROPERTY_ID_TYPOGRAPHIC_FAMILY_NAME, _familyName.data(), nullptr },
+            DWRITE_FONT_PROPERTY{ DWRITE_FONT_PROPERTY_ID_WIN32_FAMILY_NAME, _familyName.data(), nullptr },
+            DWRITE_FONT_PROPERTY{ DWRITE_FONT_PROPERTY_ID_FULL_NAME, _familyName.data(), nullptr }
+        };
+
+        Microsoft::WRL::ComPtr<IDWriteFontSet1> filteredSet1;
+        THROW_IF_FAILED(set1->GetFilteredFonts(fontProperties.data(), gsl::narrow_cast<UINT32>(fontProperties.size()), true, filteredSet1.GetAddressOf()));
+
+        if (filteredSet1->GetFontCount())
+        {
+            Microsoft::WRL::ComPtr<IDWriteFontFace5> fontFace5;
+            THROW_IF_FAILED(filteredSet1->CreateFontFace(0, fontFace5.GetAddressOf()));
+
+            // Retrieve metrics in case the font we created was different than what was requested.
+            _weight = fontFace5->GetWeight();
+            _stretch = fontFace5->GetStretch();
+            _style = fontFace5->GetStyle();
+
+            // Dig the family name out at the end to return it.
+            Microsoft::WRL::ComPtr<IDWriteLocalizedStrings> familyNamesLocalized;
+            THROW_IF_FAILED(fontFace5->GetFamilyNames(familyNamesLocalized.GetAddressOf()));

```

Comment:
This code works fine as-is. Another option, if you have the full user language list available (I forget which Windows API returns that 🤔) is to just pass the whole semicolon delimited to list to [GetPropertyValues](https://docs.microsoft.com/en-us/windows/win32/api/dwrite_3/nf-dwrite_3-idwritefontset-getpropertyvalues(dwrite_font_property_id_wcharconst_idwritestringlist)) (e.g. "ja-jp;fr-ca;en-us"), and it returns the string to you in priority order, without you needing to do the explicit fallback to en-us and then index 0. Using `DWRITE_FONT_PROPERTY_ID_WEIGHT_STRETCH_STYLE_FAMILY_NAME` is equivalent to `GetFamilyNames`, assuming you used `DWRITE_FONT_FAMILY_MODEL_WEIGHT_STRETCH_STYLE` when creating/getting the system font collection, which is the default family model for compatibility (compatible with WPF/CSS).